### PR TITLE
Added support for Chrome V3 and fixed issues with Ancestry site

### PR DESCRIPTION
**Pull request by mamsterla**
This is the code to handle the issues that made Ancestry stop working.  It also includes important work to update for Chrome V3 standards.

### CODE REVIEW

File: tmp.txt [hunk 1]
```diff
@@ -0,0 +1,704 @@
+
+<!DOCTYPE html>
```
Comment (GuyKh) [inline tmp.txt:2]: @mamsterla you sure this is needed?

File: collections/ancestrynew.js [hunk 2]
```diff
@@ -79,116 +79,97 @@ registerCollection({
...
-    var focusperson = par.find(".userCardTitle").text();
-    var focusdaterange = par.find(".userCardSubTitle").text().replace("&ndash;", " - ");
+    // TODO - figure out where the parsing issues are
+    var personCard = await readPersonCard(htmlstring);
+
+    // get personFacts
+    var pfStart = htmlstring.indexOf("researchData = ");
+    var pfEnd = htmlstring.indexOf("};", pfStart);
+
+    var personFacts = JSON.parse(htmlstring.substring(pfStart + 14, pfEnd + 1));
```
Comment (GuyKh) [inline collections/ancestrynew.js:116]: Be careful with these numeric increments, if Ancestry would decide to add another attribute to their HTML tag (i.e. script tag), you're toast and it's broken.
It's much better, IMO, to find the first `var x = {` and start from the `{`

File: collections/ancestrynew.js [hunk 3]
```diff
@@ -79,116 +79,97 @@ registerCollection({
...
+                profiledata["baptism"] = setFactData(fact);
+                break;
+            case 'Burial':
+                profiledata["burial"] = setFactData(fact);
+                if (fact.Date) burialdtflag = true;
+                if (fact.Place) buriallcflag = true;
+                break;
+            case 'Death':
+                profiledata["death"] = setFactData(fact);
+                break;
```
Comment (GuyKh) [inline collections/ancestrynew.js:159]: missing `deathdtflag=true` here

File: collections/ancestrynew.js [hunk 4]
```diff
@@ -222,60 +197,62 @@ function parseAncestryNew(htmlstring, familymembers, relation) {
...
+    for(var i = 0; i < personFacts.ResearchFamily.Mothers.length; i++) {
+        var mother = personFacts.ResearchFamily.Mothers[i];
+        if (familymembers) {
+            if (exists(mother.ClickUrl)) {
+                await getAncestryNewTreeFamily(famid++, mother.Id, mother.FullName.trim(), "mother", mother.ClickUrl);
+            }
+        } else if (exists(relation.title)) {
+            if (isChild(relation.tite)) {
+                if (personCard.personId !== mother.Id) {
+                    childlist[relation.proid] = $.inArray(mother.Id, unionurls);
```
Comment (GuyKh) [inline collections/ancestrynew.js:243]: I know this is nitpicking, but @mamsterla @eljeffeg perhaps we should move away from JQuery to standard JS libraries - e.g. `unionurls.includes(mother.id)`?

This is a far bigger task to do than this, but I'm just suggesting

**Comment by GuyKh** [comment]
Overall, this looks good! Well done @mamsterla!
I have to mention two parts here - that the V3 manifest and the Ancestry part are intertwined here.
I'd separate the two into two PRs for easier tracing of what went wrong etc.

File: collections/ancestrynew.js (hunk 4 above)
Comment (eljeffeg) [reply collections/ancestrynew.js:243]: No objection if someone wants to spend the time to convert it.

**Pull request merged**